text/0000-cargo-workspace.md

```diff
@@ -0,0 +1,345 @@
+- Feature Name: N/A
+- Start Date: 2015-09-15
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Improve Cargo's story around multi-crate single-repo project management by
+introducing the concept of workspaces. All packages in a workspace will share
+`Cargo.lock` and an output directory for artifacts.
+
+# Motivation
+
+A common method to organize a multi-crate project is to have one
+repository which contains all of the crates. Each crate has a corresponding
+subdirectory along with a `Cargo.toml` describing how to build it. There are a
+number of downsides to this approach, however:
+
+* Each sub-crate will have its own `Cargo.lock`, so it's difficult to ensure
+  that the entire project is using the same version of all dependencies. This is
+  desired as the main crate (often a binary) is often the one that has the
+  `Cargo.lock` "which counts", but it needs to be kept in sync with all
+  dependencies.
+
+* When building or testing sub-crates, all dependencies will be recompiled as
+  the target directory will be changing as you move around the source tree. This
+  can be overridden with `build.target-dir` or `CARGO_TARGET_DIR`, but this
+  isn't always convenient to set.
+
+Solving these two problems should help ease the development of large Rust
+projects by ensuring that all dependencies remain in sync and builds by default
+use already-built artifacts if available.
+
+# Detailed design
+
+Cargo will grow the concept of a **workspace** for managing repositories of
+multiple crates. Workspaces will then have the properties:
+
+* A workspace can contain multiple local crates.
```

Comment:
Could "multiple" perhaps better be replaced with "zero or more" (if a workspace with zero local crates is not forbidden)?